packages/date-picker/src/shared/components/DateInput/DateInputBox/DateInputBox.spec.tsx

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import { jest } from '@jest/globals';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { Month } from '../../../constants';
+import { SegmentRefs } from '../../../hooks';
+import { newUTC } from '../../../utils';
+import {
+  DatePickerProvider,
+  DatePickerProviderProps,
+  defaultDatePickerContext,
+} from '../../DatePickerContext';
+import { DateInputSegmentChangeEventHandler } from '../DateInputSegment/DateInputSegment.types';
+
+import { DateInputBox, type DateInputBoxProps } from '.';
+
+const renderDateInputBox = (
+  props?: Omit<DateInputBoxProps, 'segmentRefs'>,
+  context?: Partial<DatePickerProviderProps>,
+) => {
+  const segmentRefsMock: SegmentRefs = {
+    day: React.createRef(),
+    month: React.createRef(),
+    year: React.createRef(),
+  };
+
+  const result = render(
+    <DatePickerProvider {...defaultDatePickerContext} {...context}>
+      <DateInputBox
+        {...props}
+        value={props?.value ?? null}
+        segmentRefs={segmentRefsMock}
+      />
+    </DatePickerProvider>,
+  );
+
+  const dayInput = result.container.querySelector(
+    'input[aria-label="day"]',
+  ) as HTMLInputElement;
+  const monthInput = result.container.querySelector(
+    'input[aria-label="month"]',
+  ) as HTMLInputElement;
+  const yearInput = result.container.querySelector(
+    'input[aria-label="year"]',
+  ) as HTMLInputElement;
+
+  if (!(dayInput && monthInput && yearInput)) {
+    throw new Error('Some or all input segments are missing');
+  }
+
+  return { ...result, dayInput, monthInput, yearInput };
+};
+
+describe('packages/date-picker/shared/date-input-box', () => {
+  const onChange = jest.fn<DateInputSegmentChangeEventHandler>();
+
+  const testContext: Partial<DatePickerProviderProps> = {
+    dateFormat: 'iso8601',
+    timeZone: 'UTC',
+  };
+
+  afterEach(() => {
+    onChange.mockClear();
+  });
+
+  describe('Rendering', () => {
+    describe.each(['day', 'month', 'year'])('%p', segment => {
+      test('renders the correct aria attributes', () => {
+        const result = renderDateInputBox();
+        const input = result.getByLabelText(segment);
+
+        // each segment has appropriate aria label
+        expect(input).toHaveAttribute('aria-label', segment);
+      });
+    });
+
+    describe('renders segments in the correct order', () => {
+      test('iso8601', () => {
+        const result = renderDateInputBox(undefined, { dateFormat: 'iso8601' });
+        const segments = result.getAllByRole('spinbutton');
+        expect(segments[0]).toHaveAttribute('aria-label', 'year');
+        expect(segments[1]).toHaveAttribute('aria-label', 'month');
+        expect(segments[2]).toHaveAttribute('aria-label', 'day');
+      });
+
+      test('en-US', () => {
```

Comment:
should we test for every region that we support?

Reply:
We're testing the one's we've explicitly stated we're supporting. All others _should_ work, but there may be some small bugs here-&-there